htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxViewMethodReturnValueHandler.java

```diff
@@ -0,0 +1,89 @@
+package io.github.wimdeblauwe.htmx.spring.boot.mvc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.core.MethodParameter;
+import org.springframework.lang.Nullable;
+import org.springframework.ui.ModelMap;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodReturnValueHandler;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.View;
+import org.springframework.web.servlet.ViewResolver;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Handles return values that are of type {@link HtmxView}.
+ */
+public class HtmxViewMethodReturnValueHandler implements HandlerMethodReturnValueHandler {
+
+    private final ViewResolver viewResolver;
+    private final LocaleResolver localeResolver;
+
+    public HtmxViewMethodReturnValueHandler(ViewResolver viewResolver, LocaleResolver localeResolver) {
+        this.viewResolver = viewResolver;
+        this.localeResolver = localeResolver;
+    }
+
+    @Override
+    public boolean supportsReturnType(MethodParameter returnType) {
+        return HtmxView.class.isAssignableFrom(returnType.getParameterType());
+    }
+
+    @Override
+    public void handleReturnValue(Object returnValue,
+                                  MethodParameter returnType,
+                                  ModelAndViewContainer mavContainer,
+                                  NativeWebRequest webRequest) throws Exception {
+
+        if (returnValue instanceof HtmxView htmxView) {
```

Comment:
This `instanceof` assignment requires Java 14, right? I like it!

Reply:
It's called "Pattern Matching for instanceof" and available since Java 16.

https://openjdk.org/jeps/394